chore: gov docs

just-mitch opened the pull request:
Adds a bunch of clarifying comments to Governance.sol, Registry.sol, and ProposalLib.sol.

Adds a `Droppable` state to the Proposal state enum.

Renames `voteDifferential` to `requiredYeaMargin`. 

Renames `nea` to `nay`. 

Renames Proposal.state to Proposal.cachedState (see comments for reasoning).

just-mitch committed:
chore: gov docs

just-mitch commented:
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/15678?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#15678** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/15678?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/15678?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#15294** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/15294?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

LHerskind reviewed (commented):
Very neat. Have a few pretty small nits.

just-mitch commented on l1-contracts/src/governance/Governance.sol:
```diff
@@ -41,30 +124,98 @@ contract Governance is IGovernance {
+  /**
+   * @dev Withdrawals that have been initiated.
+   *
+   * `withdrawals` is only updated during `initiateWithdraw`, and `finaliseWithdraw`.
```
Thanks

just-mitch committed:
chore: empire base renames and docs (#15750)

Lot of comments on EmpireBase, GovernanceProposer, and GSEPayload, and
introduces some renames.

Generally, on the EmpireBase now, the IEmperor is used to determine the
current "signaler", who may "signal" on a "payload"; I want to preserve
"proposal" and "vote" for governance, i.e. when a payload is actually
`propose`ed to Governance.sol, it becomes part of a "proposal".

Further, the `_execute` virtual function is renamed to
`_handleRoundWinner`, so you don't "executeProposal" on the EmpireBase
anymore, instead you "submitRoundWinner" (which calls
`_handleRoundWinner`).

Also, in round accounting, `lastVote` is now `lastSignalSlot`, and
`leader` is `payloadWithMostSignals`.

just-mitch committed:
fix flake in fuzzer test
